xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiRequest.java

```diff
@@ -22,11 +22,30 @@
 import javax.servlet.http.Cookie;
 import javax.servlet.http.HttpServletRequest;
 
+import org.xwiki.stability.Unstable;
+import org.xwiki.user.GuestUserReference;
+import org.xwiki.user.UserReference;
+
 public interface XWikiRequest extends HttpServletRequest
 {
     String get(String name);
 
     HttpServletRequest getHttpServletRequest();
 
     Cookie getCookie(String cookieName);
+
+    /**
+     * @return the user that holds the responsibility, in terms of access rights, for the submitted data and the changes
+     *         triggered by this request. By default the effective author is the user that gets authenticated with the
+     *         information provided by this request. If the request doesn't indicate an effective author and no user is
+     *         authenticated then the {@link GuestUserReference} is returned.
+     * @since 15.10.11
+     * @since 16.4.1
+     * @since 16.5.0RC1
+     */
+    @Unstable
+    default UserReference getEffectiveAuthor()
+    {
+        return GuestUserReference.INSTANCE;
```

Comment:
Not sure this is the most retro compatible default. What about returning null (as in "no idea what is the user associated with the request, rely on other fallbacks", or an emty Optional<UserReference>) instead ?

Note that there is currently nothing preventing XWikiServletRequest#getEffectiveAuthor() to return null so it could happen, and it will definitely happen a lot in the case of a XWikiServletRequestStub.

Reply:
You're right. I made it return an optional user reference, with an empty optional by default (i.e. the request doesn't indicate any effective author).